src/simulator/network.c

```diff
@@ -181,3 +179,47 @@ network_get_layer_spikes(State* state, Network* network, u32 i) {
     error:
     return NULL;
 }
+
+
+// LEARNING
+internal void
+network_learning_step(Network* network, Inputs* inputs, u32 time,
+                      MemoryArena* storage, ThreadPool* pool) {
+    TIMING_COUNTER_START(NETWORK_LEARNING_STEP);
+    
+    check(network != NULL, "network is NULL");
+    check(network->n_in_layers == inputs->n_inputs,
+          "network->n_in_layers is %u, inputs->n_inputs is %u, should be equal",
+          network->n_in_layers, inputs->n_inputs);
+    check(storage != NULL, "storage is NULL");
+    check(pool != NULL, "pool is NULL");
+    
+    Layer* layer = NULL;
+    Input* input = NULL;
+    u32 i = 0;
+    
+    // NOTE: Assume that the order of inputs are the same as the order of input layers in
+    // NOTE: the network
+    for (i = 0; i < inputs->n_inputs; ++i) {
+        input = inputs->inputs + i;
+        layer = network->layers[i];
+        
+        if (input->type == INPUT_SPIKES)
```

Comment:
I think you can use modes again here, its simple and it removes some duplicateion